charts/loki-distributed/templates/query-frontend/service-query-frontend.yaml

```diff
@@ -10,6 +10,7 @@ metadata:
 spec:
   type: ClusterIP
   clusterIP: None
+  publishNotReadyAddresses: true
```

Comment:
We need to add this after adding a readiness probe to the query frontend, as otherwise the dns name is not added to the service and no querier can connect to the query frontend as defined at the end of the section https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-hostname-and-subdomain-fields